WcaOnRails/app/controllers/api/v0/users_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+class Api::V0::UsersController < Api::V0::ApiController
+  def me
+    if current_user
+      if stale?(current_user)
+        # Also include the users current prs so we can handle qualifications on the Frontend
+        show_user(current_user, show_rankings: true)
+      end
+    else
+      render status: :unauthorized, json: { error: "Please log in" }
+    end
+  end
+
+  def show_user_by_id
+    user = User.find_by_id(params[:id])
+    show_user(user)
+  end
+
+  def show_user_by_wca_id
+    user = User.find_by_wca_id(params[:wca_id])
+    show_user(user)
+  end
+
+  def permissions
+    if current_user
+      if stale?(current_user)
```

Comment:
Not sure if this `stale` call catches all relevant information for permissions... For example a `team_member` association to the "banned" team might change, but the `User` model itself didn't. Then I think this will return outdated responses.

Reply:
I tested it out in on staging and it correctly showed it after a refresh (actually two refreshs?)